### PR TITLE
Add UMAP visualization support

### PR DESCRIPTION
**Pull request by gokceneraslan**
I added UMAP support for visualization \o/ Here is how MNIST "single cells" look like:

image

image

I'm not so familiar with code sytle and variable naming etc. yet, and I haven't fully tested things like additional umap kwargs ~and 3d visuazliation~ etc. but let's keep PR here and resolve things along the way.

### CODE REVIEW

**Comment by gokceneraslan** [comment]
OK, this was easier than I thought:

image

File: docs/references.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
        <style type="text/css">
        .label {
-       }
```
Comment (gokceneraslan) [inline docs/references.rst:6]: Sorry about removing whitespace. I just couldn't resist 🙈

**Comment by gokceneraslan** [comment]
More examples:

image

image

**Comment by falexwolf** [comment]
Nice, nice, nice! :smile: Great!

File: docs/references.rst (hunk 1 above)
Comment (falexwolf) [reply docs/references.rst:6]: sure! there should be no whitespace here. maybe some relict from when this was all markdown and three whitespaces meant a line break

**Comment by terooatt** [comment]
gokceneraslan 

Thank you very much for the example you have provided with umap utilization. It looks pretty straight forward to use. If I had initially combined different samples and generate a umap plot do you know how you could visualize one sample while keeping the overall structure of the umap pot?

**Pull request merged**